### PR TITLE
ARC-1599 - Create "create branch" basic form inputs

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- View updated for Create a Branch.
- Added 3 inputs
  - 2 [select2](https://aui.atlassian.com/aui/latest/docs/auiselect2.html) inputs
  - 1 simple input 

**Why**
- Following the designs

**How has this been tested?**  
- Locally
- Staging

**Screenshot**
Screen Shot 2022-09-02 at 9 28 18 am
Screen Shot 2022-09-02 at 9 28 12 am
Screen Shot 2022-09-02 at 9 27 57 am

### CODE REVIEW

File: views/github-create-branch.hbs [hunk 1]
```diff
@@ -21,15 +21,61 @@
...
-    Create a Branch
+    <div class="gitHubCreateBranch__header">Create a Branch</div>
+    <div class="gitHubCreateBranch__subHeader">
+      Creating a branch for
+      <!--TODO: Need to add icons based on the type of issue(story/task/bug) -->
+      <a href="#">KEY-1</a>
+    </div>
   </div>
+
+  <div class="gitHubCreateBranch__optionContainer">
```
Comment (rachellerathbone) [inline views/github-create-branch.hbs:39]: Don't we want to hide this unless the customer has both?

Comment (rachellerathbone) [reply views/github-create-branch.hbs:39]: Actually, if we have to go through auth before this do we even need this? This is similar to the connect an org page in the sense that we've already done the auth handshake, then we render what the user has access to for that server/app/org.

Comment (joshkay10) [reply views/github-create-branch.hbs:39]: Yeah, feel free to drop this section for now, we are still figuring out the ins and outs when it comes to authentication.
Has its own story to deal with it.

File: src/routes/github/create-branch/github-create-branch-get.ts [hunk 2]
```diff
@@ -11,13 +16,15 @@ export const GithubCreateBranchGet = async (req: Request, res: Response, next: N
 		return next(new Error(Errors.MISSING_GITHUB_TOKEN));
 	}
 
-
 	res.render("github-create-branch.hbs", {
 		csrfToken: req.csrfToken(),
 		jiraHost,
 		nonce: res.locals.nonce,
-		title: "Create a Branch"
+		title: "Create a Branch",
```
Comment (joshkay10) [inline src/routes/github/create-branch/github-create-branch-get.ts:23]: Do we need to pass the title here?

File: views/github-create-branch.hbs [hunk 3]
```diff
@@ -21,15 +21,61 @@
 <input type="hidden" id="_csrf" name="_csrf" value="{{csrfToken}}" />
 <input type="hidden" id="jiraHost" name="jiraHost" value="{{jiraHost}}" />
 
-{{> navigation hideBackButton=true }}
+{{> navigation }}
```
Comment (joshkay10) [inline views/github-create-branch.hbs:24]: Turns out we can just remove this completely

File: views/github-create-branch.hbs (hunk 1 above)
Comment (krazziekay) [reply views/github-create-branch.hbs]: This has been removed for this task.

File: src/routes/github/create-branch/github-create-branch-get.ts (hunk 2 above)
Comment (krazziekay) [reply src/routes/github/create-branch/github-create-branch-get.ts]: Its not necessary I guess, was using it before for the dummy page. I've removed this

File: views/github-create-branch.hbs (hunk 3 above)
Comment (krazziekay) [reply views/github-create-branch.hbs]: Removed

File: views/github-create-branch.hbs [hunk 4]
```diff
@@ -21,15 +21,47 @@
...
 
-{{> navigation hideBackButton=true }}
-
 <section class="gitHubCreateBranch">
   <div class="headerImage">
     <img src="/public/assets/jira-and-github.png" alt="Jira and GitHub logos" />
   </div>
   <div class="gitHubCreateBranch__title">
-    Create a Branch
+    <div class="gitHubCreateBranch__header">Create a Branch</div>
```
Comment (rachellerathbone) [inline views/github-create-branch.hbs:29]: Just checked the designs, this should be Create GitHub branch

Screen Shot 2022-09-01 at 4 20 09 pm

File: views/github-create-branch.hbs [hunk 5]
```diff
@@ -21,15 +21,47 @@
...
+    <div class="gitHubCreateBranch__subHeader">
+      Creating a branch for
+      <!--TODO: Need to add icons based on the type of issue(story/task/bug) -->
+      <a href="#">KEY-1</a>
+    </div>
   </div>
+
+  <form id="createBranchForm" class="aui top-label">
+    <div class="field-group">
```
Comment (rachellerathbone) [inline views/github-create-branch.hbs:38]: Looks like we're missing the GitHub Enterprise Server instance input for server customers.

Comment (krazziekay) [reply views/github-create-branch.hbs:38]: Added the field, but haven't added the condition to display it, which I assume will be done in a separate ticket.

File: views/github-create-branch.hbs (hunk 4 above)
Comment (krazziekay) [reply views/github-create-branch.hbs:29]: Updated

File: views/github-create-branch.hbs [hunk 6]
```diff
@@ -21,15 +21,56 @@
...
   <div class="headerImage">
     <img src="/public/assets/jira-and-github.png" alt="Jira and GitHub logos" />
   </div>
   <div class="gitHubCreateBranch__title">
-    Create a Branch
+    <div class="gitHubCreateBranch__header">Create GitHub Branch</div>
+    <div class="gitHubCreateBranch__subHeader">
+      Creating a branch for
+      <!--TODO: Need to add icons based on the type of issue(story/task/bug) -->
+      <a href="#">KEY-1</a>
```
Comment (joshkay10) [inline views/github-create-branch.hbs:33]: Nice, the link can be added in story ARC-1598, should be a quick win there :)

File: views/github-create-branch.hbs [hunk 7]
```diff
@@ -21,15 +21,56 @@
...
+    <div class="field-group">
+      <aui-label for="repo">Repository</aui-label>
+      <select id="ghRepo" name="repo" class="full-width-field" placeholder="Select a repository">
+        {{#each repos}}
+          <option value="{{id}}">{{org}} / {{name}}</option>
+        {{/each}}
+      </select>
+    </div>
+    <div class="field-group">
+      <aui-label for="branch">Repository</aui-label>
```
Comment (joshkay10) [inline views/github-create-branch.hbs:56]:
```suggestion
      <aui-label for="branch">Branch from</aui-label>
```

File: views/github-create-branch.hbs [hunk 8]
```diff
@@ -21,15 +21,56 @@
...
+      <select id="ghParentBranch" name="parentBranch" class="full-width-field" placeholder="Select a branch">
+        {{#each branches}}
+          <option value="{{id}}">{{name}}</option>
+        {{/each}}
+      </select>
+    </div>
+    <div class="field-group">
+      <label for="branch-name">Branch name</label>
```
Comment (joshkay10) [inline views/github-create-branch.hbs:64]: Using `aui-label` for the other fields and the font size looks a little bit smaller here.

File: static/js/github-create-branch.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+$(document).ready(() => {
+  $("#ghServers").auiSelect2();
+  $("#ghRepo").auiSelect2();
+  $("#ghParentBranch").auiSelect2();
+
+  $(".gitHubCreateBranch__option").on("click", event => {
```
Comment (joshkay10) [inline static/js/github-create-branch.js:6]: Is this still needed? Couldn't find .gitHubCreateBranch__option? could have just missed it tho lolol

File: views/github-create-branch.hbs (hunk 7 above)
Comment (krazziekay) [reply views/github-create-branch.hbs:56]: Updated.

File: static/js/github-create-branch.js (hunk 9 above)
Comment (krazziekay) [reply static/js/github-create-branch.js:6]: Ya missed out this one. This has been removed.

**Review by rachellerathbone — APPROVED** [review]

**Pull request merged**